### PR TITLE
Use node to run tests

### PR DESCRIPTION
**Pull request by Nathan-SL**
- Avoids 0 passing tests message by mocha
- Removes flag for account-contract from `starknet-verify` test
- Update exec function to break on exit with status code different from 0

## Usage related changes

-   NA

## Development related changes

-   Tests are launched by `node` instead of `mocha`

## Checklist:

-   [x] Formatted the code
-   [x] No linter errors + tried to avoid introducing linter warnings
-   [x] Performed a self-review of the code
-   [x] Rebased to the last commit of the target branch (or merged it into my branch)
-   [x] Updated the `test` directory (with a test case consisting of `network.json`, `hardhat.config.ts`, `check.sh`)
-   [ ] All tests are passing (for external contributors who don't have access to the CI/CD pipeline)

### CODE REVIEW

File: scripts/test.sh [hunk 1]
```diff
@@ -72,7 +72,7 @@ function iterate_dir() {
         # check if test_case/check.ts exists
         if [ -f "$test_case/check.ts" ]; then
             # run the test
-            NETWORK="$network" npx mocha -r ts-node/register "$test_case/check.ts" && success=$((success + 1)) || echo "Test failed!"
+            NETWORK="$network" node -r ts-node/register "$test_case/check.ts" && success=$((success + 1)) || echo "Test failed!"
```
Comment (FabijanC) [inline scripts/test.sh:75]: I think this can be `npx ts-node ...`

Comment (Nathan-SL) [reply scripts/test.sh]: https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/259/commits/c5648b478f26cef526334f74981f76d0ab238286

File: src/types.ts [hunk 2]
```diff
@@ -756,6 +756,7 @@ export class StarknetContract {
     decodeEvents(events: starknet.Event[]): DecodedEvent[] {
         const decodedEvents: DecodedEvent[] = [];
         for (const event of events) {
+            if (parseInt(event.from_address, 16) !== parseInt(this.address, 16)) continue;
```
Comment (FabijanC) [inline src/types.ts:759]: The formatter agrees with this?

**Review by FabijanC — CHANGES_REQUESTED** [review]

File: src/types.ts (hunk 2 above)
Comment (Nathan-SL) [reply src/types.ts:759]: Yes.

File: src/types.ts [hunk 3]
```diff
@@ -768,6 +769,10 @@ export class StarknetContract {
             decodedEvents.push({ name: eventSpecification.name, data: adapted });
         }
 
+        if (decodedEvents.length === 0) {
+            const msg = `No events were decoded. You might be using a wrong contract. ${this.abiPath}`;
```
[the remark this answers is not in the document]
Comment (Nathan-SL) [reply src/types.ts]: https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/259/commits/46a4d605fb236381962bb74fc96e591ad634a590

File: test/utils/cli-functions.ts [hunk 4]
```diff
@@ -1,48 +1,58 @@
...
-const exec = (cmd: string, silent?: boolean) => {
-    silent = silent || false;
-    return shell.exec(cmd, { silent: silent });
+export function exec(cmd: string, expectFailure?: boolean) {
+    expectFailure = expectFailure || false;
+    const result = shell.exec(cmd, { silent: expectFailure });
+    if (!expectFailure) {
+        assertEqual(result.code, 0, `Command ${cmd} failed.\n${result.stderr}`);
+    }
+
```
[the remark this answers is not in the document]
Comment (Nathan-SL) [reply test/utils/cli-functions.ts:12]: https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/259/commits/683c3c37ff6d70bbb725c91f9554271e1998b087

File: test/utils/deploy-funded-account.ts [hunk 5]
```diff
@@ -21,12 +21,18 @@ export async function deployFundedAccount(url = DEVNET_URL) {
...
     console.log(`Funding account ${accountAddress} on ${network}.`);
-    await axios.post(`${url}/mint`, {
+    const data = JSON.stringify({
         address: accountAddress,
-        amount: 1000000000000000000n,
+        amount: 10 ** 18,
         lite: true
     });
 
+    await axios.post(`${url}/mint`, data, {
```
[the remark this answers is not in the document]
Comment (Nathan-SL) [reply test/utils/deploy-funded-account.ts]: https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/259/commits/9b236d0f292b0abcab45b011e3c3d5f3bb2aa2fa great catch!

File: test/utils/cli-functions.ts [hunk 6]
```diff
@@ -1,48 +1,58 @@
 import shell from "shelljs";
+import { assertEqual } from "./utils";
 
-const exec = (cmd: string, silent?: boolean) => {
-    silent = silent || false;
-    return shell.exec(cmd, { silent: silent });
+export function exec(cmd: string, expectFailure?: boolean) {
```
[the remark this answers is not in the document]
Comment (Nathan-SL) [reply test/utils/cli-functions.ts]: https://github.com/Shard-Labs/starknet-hardhat-plugin/commit/683c3c37ff6d70bbb725c91f9554271e1998b087

File: test/utils/cli-functions.ts [hunk 7]
```diff
@@ -1,48 +1,58 @@
...
+    expectFailure = expectFailure || false;
+    const result = shell.exec(cmd, { silent: expectFailure });
```
[the remark this answers is not in the document]
Comment (Nathan-SL) [reply test/utils/cli-functions.ts:5]: For most tests, we do want the output to be printed which would save us from having the result of execution be stored in a variable and then printing the stdout of the result.

**Review by FabijanC — APPROVED** [review]

**Pull request merged**